test/integration_test/basic_test.go

```diff
@@ -16,9 +18,58 @@ import (
 )
 
 func TestStorageClusterBasic(t *testing.T) {
+	testutil.TestSpecPath = ""
 	t.Run("InstallWithAllDefaults", testInstallWithAllDefaults)
 	t.Run("NodeAffinityLabels", testNodeAffinityLabels)
 	t.Run("Upgrade", testUpgrade)
+	t.Run("InstallMetricsCollector", testInstallMetricsCollector)
+}
+
+func testInstallMetricsCollector(t *testing.T) {
+	secret := testutil.GetExpectedSecret(t, "/specs/pure-telemetry-cert.yaml")
+	require.NotNil(t, secret)
+
+	_, err := coreops.Instance().GetSecret(component.TelemetryCertName, pxNamespace)
+	require.True(t, err == nil || errors.IsNotFound(err))
+
+	if errors.IsNotFound(err) {
+		secret, err = coreops.Instance().CreateSecret(secret)
+		require.NoError(t, err)
+	} else {
+		secret, err = coreops.Instance().UpdateSecret(secret)
+		require.NoError(t, err)
+	}
+
+	// Deploy portworx with telemetry set to true
+	cluster, err := constructDefaultStorageCluster()
+	require.NoError(t, err)
+
+	cluster.Spec.Monitoring = &op_corev1.MonitoringSpec{
+		Telemetry: &op_corev1.TelemetrySpec{
+			Enabled: true,
+		},
+	}
+
+	cluster, err = createStorageCluster(cluster)
+	require.NoError(t, err)
+
+	// Validate metrics collector pod becomes ready
+	err = testutil.ValidatePods(
+		cluster.Namespace,
+		map[string]string{"role": "realtime-metrics-collector"},
+		defaultValidateDeployTimeout,
+		defaultValidateDeployRetryInterval)
+	require.NoError(t, err)
```

Comment:
We probably don't need this. Should we call `testutil.ValidateStorageCluster()` instead to validate the entire cluster. This function validates all the components as well. We will have to add metrics collector to the validation.

Reply:
ValidateStorageCluster does not verify every component, it only verifies if storagecluster is online. let's let each component verify itself

Reply:
ValidateStorageCluster check all the components. It not only checks for the deployment, but also check if they have the right images. You should add the metrics collector component to that function. 
After that if you want, you can call just the ValidateMetricsCollector part out of it in this test.